pkg/splunk/client/azureblobclient.go

```diff
@@ -108,126 +207,242 @@ func NewAzureBlobClient(ctx context.Context, bucketName string, storageAccountNa
-	scopedLog := reqLogger.WithName("InitAzureBlobClientSession")
+	scopedLog := reqLogger.WithName("updateHttpRequestHeaderWithSecrets")
+
+	scopedLog.Info("Updating Azure Http Request with secrets")
 
-	scopedLog.Info("Initializing  AzureBlob Service for apps", "appAzureBlobEndPoint", appAzureBlobEndPoint)
-	var azureBlobClient SplunkAzureBlobClientTemp
+	// Update httpRequest header with data and version
+	httpRequest.Header[headerXmsDate] = []string{time.Now().UTC().Format(http.TimeFormat)}
+	httpRequest.Header[headerXmsVersion] = []string{azureHTTPHeaderXmsVersion}
+
+	// Get HMAC signature using storage account name and secret access key
+	stringToSign, err := buildStringToSign(*httpRequest, client.StorageAccountName)
+	if err != nil {
+		scopedLog.Error(err, "Azure Blob with secrets Failed to build string to sign")
+		return err
+	}
+	decodedAccountKey, err := base64.StdEncoding.DecodeString(client.SecretAccessKey)
+	if err != nil {
+		// failed to decode
+		scopedLog.Error(err, "Azure Blob with secrets failed to decode accountKey")
+		return err
+	}
+	signature := ComputeHMACSHA256(stringToSign, decodedAccountKey)
+	authHeader := strings.Join([]string{"SharedKey ", client.StorageAccountName, ":", signature}, "")
 
-	// TODO target for next sprint CSPL XXXX List and download azure object
-	// This part need to be developed when the Azure blob rest handlers
-	// are built.
+	// Update httpRequest header with the HMAC256 signature
+	httpRequest.Header[headerAuthorization] = []string{authHeader}
 
-	// var err error
+	return nil
+}
 
-	// Enforcing minimum version TLS1.2
-	// tr := &http.Transport{
-	// 	TLSClientConfig: &tls.Config{
-	// 		MinVersion: tls.VersionTLS12,
-	// 	},
-	// }
-	// tr.ForceAttemptHTTP2 = true
-	// httpClient := http.Client{Transport: tr}
+// Update http request header with IAM info
+func updateAzureHTTPRequestHeaderWithIAM(ctx context.Context, client *AzureBlobClient, httpRequest *http.Request) error {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("updateHttpRequestHeaderWithIAM")
 
-	// config := &Config{
-	// 	MaxRetries: 3,
-	// 	HTTPClient: httpClient,
-	// }
+	scopedLog.Info("Updating Azure Http Request with IAM")
 
-	// azureBlobClient, err = NewAzureSession(ctx, appAzureBlobEndPoint, storageAccountName, secretAccessKey, config)
-	// if err != nil {
-	// 	scopedLog.Info("Error creating new AzureBlob Client Session", "err", err)
-	// 	return nil
-	// }
+	// Create http request to retrive IAM oauth token from metadata URL
+	oauthRequest, err := http.NewRequest("GET", azureTokenFetchURL, nil)
+	if err != nil {
+		scopedLog.Error(err, "Azure Blob Failed to create new token request")
+		return err
+	}
 
-	azureBlobClient = SplunkAzureBlobClientTemp{client: http.Client{}}
-	return &azureBlobClient
+	// Mark metadata flag
+	oauthRequest.Header.Set("Metadata", "true")
 
-}
+	// Create raw query for http request
+	values := oauthRequest.URL.Query()
+	values.Add("api-version", "2018-02-01")
```

Comment:
where is this date coming from?

Reply:
Good point. Added reference for this date in the names.go file as well as replaced this date with a newer/latest date that azure supports